app/controllers/api/orders_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Api::OrdersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :is_order_confirmed, only: [:update]
+
+  def create
+    product = Product.find(params[:product_id])
+    order = current_user.orders.create
+    order.items.create(product: product)
+    if order.persisted?
+      render json: {
+        message: 'The item was added to your order',
+        order: {
+          id: order.id,
+          items: order.products
+        }
+      }, status: 201
+    else
+      render json: {
+        message: 'Well, that went wrong..'
+      }, status: 422
+    end
+  end
+
+  def update
+    order = Order.find(params['order_id'])
+    
+    if params['confirmed']
+      order.update(confirmed: true)
+      render json: {
+        message: 'Your order is confirmed and will be available sort of soon!'
+      }
+    else
+      product = Product.find(params[:product_id])
+      order.items.create(product: product)
+      if order.persisted?
```

Comment:
You should have though about refactor the response into a private method to avoid code repetition from the create action